src/rtracer.js

```diff
@@ -96,7 +96,7 @@ const fastifyPlugin = (fastify, options, next) => {
     if (useFastifyRequestId) {
       requestId = requestId || request.id
     }
-    requestId = requestId || requestIdFactory()
+    requestId = requestId || (requestIdFactory ? requestIdFactory(request.raw) : uuidv1())
```

Comment:
```suggestion
    requestId = requestId || (requestIdFactory ? requestIdFactory(request) : uuidv1())
```

Reply:
The raw contains all the info one might need from the request and will be compatible between all versions of fastify.